### PR TITLE
Add new option WorkflowIdReusePolicy on creating new workflow in cli

### PR DESCRIPTION
**Pull request by yux0**
Add new option WorkflowIdReusePolicy on creating new workflow in cli

Issue: https://github.com/uber/cadence/issues/704

### CODE REVIEW

File: tools/cli/commands.go [hunk 1]
```diff
@@ -1336,3 +1340,23 @@ func getWorkflowStatus(statusStr string) s.WorkflowExecutionCloseStatus {
...
+
+func getWorkflowIdReusePolicy(definedPolicy string) *s.WorkflowIdReusePolicy {
+	if definedPolicy != "" {
+		value, err := strconv.Atoi(definedPolicy)
+		if err != nil {
+			ErrorAndExit(fmt.Sprintf("Option %v format is not supported.", FlagWorkflowIdReusePolicy), err)
+			return nil
+		}
+		policy := s.WorkflowIdReusePolicy(value)
+		for _, p := range s.WorkflowIdReusePolicy_Values() {
```
Comment (yux0) [inline tools/cli/commands.go]: Not sure if there is better way to check value in array

File: tools/cli/workflow.go [hunk 2]
```diff
@@ -143,6 +143,11 @@ func newWorkflowCommands() []cli.Command {
 					Usage: "Optional input for the workflow from JSON file. If there are multiple JSON, concatenate them and separate by space or newline. " +
 						"Input from file will be overwrite by input from command line",
 				},
+				cli.StringFlag{
+					Name: FlagWorkflowIdReusePolicyAlias,
+					Usage: "Optional input to configure if the same workflow ID is allow to use for new workflow execution (choose from 0 to 2). " +
```
Comment (vancexu) [inline tools/cli/workflow.go]: Could we explicit present all options like `0: AllowDuplicateFailedOnly, 1: AllowDuplicate, 2: RejectDuplicate` to save user another link click?

File: tools/cli/workflow.go [hunk 3]
```diff
@@ -143,6 +143,11 @@ func newWorkflowCommands() []cli.Command {
 					Usage: "Optional input for the workflow from JSON file. If there are multiple JSON, concatenate them and separate by space or newline. " +
 						"Input from file will be overwrite by input from command line",
 				},
+				cli.StringFlag{
```
Comment (vancexu) [inline tools/cli/workflow.go]: Maybe worth using IntFlag, and use IsSet() to tell difference between 0 and isSet.

File: tools/cli/commands.go (hunk 1 above)
Comment (vancexu) [reply tools/cli/commands.go]: To validate input, use `value >= 0 && value < len(s.WorkflowIdReusePolicy_Values())` then you can directly use policy. 
Other way is to use `switch policy ... `.

File: tools/cli/workflow.go (hunk 3 above)
Comment (yux0) [reply tools/cli/workflow.go]: Thanks. This is what I am looking for. Updated

File: tools/cli/workflow.go (hunk 2 above)
Comment (yux0) [reply tools/cli/workflow.go]: Done.

File: tools/cli/commands.go (hunk 1 above)
Comment (yux0) [reply tools/cli/commands.go]: Good point

**Pull request merged**